catroid/src/main/java/org/catrobat/catroid/ui/WebViewActivity.java

```diff
@@ -313,8 +318,26 @@ private boolean isWhatsappInstalled() {
 		}
 	}
 
+	@Override
+	public void onReceiveResult(int resultCode, Bundle resultData) {
+		long progress;
+		if (resultCode == Constants.UPDATE_DOWNLOAD_PROGRESS) {
+			progress = resultData.getLong(ProgressResponseBody.TAG_PROGRESS);
+			updateProgressDialog(progress);
+		} else if (resultCode == Constants.UPDATE_DOWNLOAD_SUCCESS) {
+			progress = resultData.getLong(ProgressResponseBody.TAG_PROGRESS);
+			updateProgressDialog(progress);
+			ToastUtil.showSuccess(WebViewActivity.this, "Download successful");
```

Comment:
There is a hardcoded string `"Download successful"`. you really must not do that, please have a look at how to correctly provide strings that are visible on the UI.